### PR TITLE
refactor(destination): added log and change returned error during searching objects for retl sources to debug chatpay

### PR DESCRIPTION
**Pull request by ruchiramoitra**
## Description of the change

> Chatpay having issue with searching contacts which cannot be reproduced locally. Added extra log to receive the exact error so that we can debug.

After debugging we understood the need to change the error code that is being returned to try retries for api limit rate.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/hs/util.js [hunk 1]
```diff
@@ -465,7 +465,7 @@ const getExistingData = async (inputs, destination) => {
 
     if (searchResponse.status !== 200) {
       throw new CustomError(
-        "[HS]:: rETL - Error during searching object record.",
+        `[HS]:: rETL - Error during searching object record.${searchResponse.response.message}`,
```
Comment (akashrpo) [inline v0/destinations/hs/util.js:468]:
```suggestion
        `[HS]:: rETL - Error during searching object record. ${searchResponse.response.message}`,
```

**Review by akashrpo — APPROVED** [review]

**Review by akashrpo — DISMISSED** [review]

File: v0/destinations/hs/util.js [hunk 2]
```diff
@@ -465,7 +465,7 @@ const getExistingData = async (inputs, destination) => {
 
     if (searchResponse.status !== 200) {
       throw new CustomError(
-        "[HS]:: rETL - Error during searching object record.",
+        `[HS]:: rETL - Error during searching object record. ${searchResponse.response.message}`,
```
Comment (utsabc) [inline v0/destinations/hs/util.js:468]:
```suggestion
        `[HS]:: rETL - Error during searching object record. ${searchResponse.response?.message}`,
```

**Review by utsabc — CHANGES_REQUESTED** [review]

**Review by utsabc — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

File: v0/destinations/hs/util.js [hunk 3]
```diff
@@ -465,8 +467,8 @@ const getExistingData = async (inputs, destination) => {
 
     if (searchResponse.status !== 200) {
       throw new CustomError(
-        "[HS]:: rETL - Error during searching object record.",
-        400
```
Comment (krishna2020) [inline v0/destinations/hs/util.js:469]: can we add corresponding test case too ?

**Review by yashasvibajpai — DISMISSED** [review]
Looks good. 
We can add a small supporting test case for this (non-blocking)

File: v0/destinations/hs/util.js (hunk 3 above)
Comment (ruchiramoitra) [reply v0/destinations/hs/util.js:469]: done

**Review by mihir-4116 — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

**Pull request merged**